### PR TITLE
Add IsInf to detect infinity values

### PR DESCRIPTION
**Pull request by wschin**
In addition to `NaN`, positive and negative values are also very important special values in floating-number system, so it'd be nice if `IsInf` can support them.

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -9531,6 +9531,49 @@ This version of the operator has been available since version 10 of the default
...
+      attributes.
+
+#### Version
+
+This version of the operator has been available since version 10 of the default ONNX operator set.
+
+#### Attributes
+
+<dl>
+<dt><tt>detectnegativeinfinity</tt> : int (default is 0)</dt>
```
Comment (ebarsoum) [inline docs/Changelog.md]: Use the same style for attribute naming as the rest of ONNX. The above should be 'detect_negative_infinity'.

File: docs/Changelog.md [hunk 2]
```diff
@@ -9531,6 +9531,49 @@ This version of the operator has been available since version 10 of the default
...
+<dd>(Optional) Whether map negative infinity to true. Default to 1 so that negative infinity induces true. Set this attribute to 0 if positive infinity should be mapped to false.</dd>
```
Comment (gramalingam) [inline docs/Changelog.md]: "positive" => "negative"

File: onnx/defs/tensor/defs.cc [hunk 3]
```diff
@@ -1563,13 +1563,40 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            "(Optional) Whether map positive infinity to true. Default to 1 "
+            "so that positive infinity induces true. Set this attribute to 0 "
+            "if positive infinity should be mapped to false.",
+            AttributeProto::INT,
+            static_cast<int64_t>(0))
+        .Attr(
+            "detectnegativeinfinity",
+            "(Optional) Whether map negative infinity to true. Default to 1 "
+            "so that negative infinity induces true. Set this attribute to 0 "
+            "if positive infinity should be mapped to false.",
```
Comment (gramalingam) [inline onnx/defs/tensor/defs.cc]: "positive" => "negative"

**Comment by gramalingam** [comment]
Adding a test-case to test this would be helpful.

**Comment by spandantiwari** [comment]
@wschin - This is useful functionality and will be a welcome addition to ONNX.

But have we considered an alternate design where we have a separate op for detecting `Inf` and `-Inf`, possibly named `IsInf`? 

`Inf` and `NaN` are two different floating-point numbers with totally different meaning and IEEE representation. Putting Infs under the `IsNan` op is not the best design. I am aware that we do not want to add too many ops to the spec, but I think this is one of those cases where clarity of design trumps that concern with adding another op. As example, they are separate ops in TF, Numpy, MATLAB, and PyTorch (in PyTorch there is no isinf, but isnan is just for NaNs not Infs).

**Comment by wschin** [comment]
@spandantiwari, sounds good. I will update this PR to add `IsInf` with `detect_positive` and `detect_negative` attributes.

File: docs/Changelog.md (hunk 1 above)
Comment (wschin) [reply docs/Changelog.md]: Test files are added and now the strategy becomes
- Not to touch IsNaN
- Add IsInf instead.

File: docs/Changelog.md (hunk 2 above)
Comment (wschin) [reply docs/Changelog.md]: Ok.

File: onnx/defs/tensor/defs.cc (hunk 3 above)
Comment (wschin) [reply onnx/defs/tensor/defs.cc]: Ok.

**Comment by spandantiwari** [comment]
@wschin - OK, that sounds good. Thanks for your consideration.

**Comment by spandantiwari** [comment]
> @spandantiwari, sounds good. I will update this PR to add `IsInf` with `detect_positive` and `detect_negative` attributes.

LGTM.

**Pull request merged**